frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -21,8 +23,14 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   dirty,
   namespace,
   projects,
+  showSinkAlert,
 }) => (
   <FlexForm onSubmit={handleSubmit}>
+    {showSinkAlert && (
+      <Alert variant="default" title="Event Source can not be created" isInline>
+        An event source must sink to Knative Service, but no Knative Service exist in this project
+      </Alert>
+    )}
```

Comment:
Move to separate component and have it do it's own fetching of the knative service to show the alert.

Right now this change is spread across 3 different components when it should be isolated to a single component.

Also try to use the firehose hook.

Reply:
Have fixed it and refactored code Projects are also fetched with `useK8sWatchResource`